pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -462,6 +501,66 @@ pub mod pallet {
 
 			Ok((address, incoming_msg))
 		}
+
+		fn service_outbound_message_queue(max_weight: Weight) -> Weight {
+			let mut weight_used = Weight::zero();
+
+			let mut processed_entries = Vec::new();
+
+			for ((domain, sender, message), _) in OutboundMessageQueue::<T>::iter() {
+				processed_entries.push((domain.clone(), sender.clone(), message.clone()));
+
+				let router = match DomainRouters::<T>::get(domain.clone()) {
+					Some(router) => router,
+					None => {
+						Self::deposit_event(Event::RouterNotFound {
+							domain: domain.clone(),
+						});
+
+						continue;
+					}
+				};
+
+				let res = match router.send(sender.clone(), message.clone()) {
+					Ok(r) => {
+						Self::deposit_event(Event::RouterExecutionSuccess {
+							sender: sender.clone(),
+							domain: domain.clone(),
+							message: message.clone(),
+						});
+
+						r
+					}
+					Err(e) => {
+						Self::deposit_event(Event::RouterExecutionFailure {
+							sender: sender.clone(),
+							domain: domain.clone(),
+							message: message.clone(),
+							error: e.error,
+						});
+
+						e.post_info
+					}
+				};
+
+				let weight = res
+					.actual_weight
+					.or(Some(Weight::from_parts(DEFAULT_WEIGHT_REF_TIME, 0)))
```

Comment:
PoV weight is unfortunately not negligible here. The pov weight would at least be `MaxEnodedLen(T::Domain) + MaxEnodedLen(T::AccountId) + MaxEnodedLen(T::Message)` for each iterated item.

Reply:
And this weight is always added. Not Matter if `actual_weight` is `Some` or not.

Reply:
I updated all weight calculations here.